alkos/kernel/test/tester.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef ALKOS_TEST_TESTER_HPP_
+#define ALKOS_TEST_TESTER_HPP_
+
+#include <stdint.h>
+
+enum TestType: uint64_t {
+    kStackSmashTest,
+    kLastTest,
+};
+
+#define IS_MACRO_EMPTY_(x) 0##x
```

Comment:
Do we need macros?
will a constexpr approach not work? (below snippet is genereated by copilot) 
```c++
constexpr bool isNotEmpty(const char* str) {
    return str[0] != '\0';
}

constexpr bool isValidTestType(TestType type) {
    return type < TestType::kLastTest;
}

constexpr void verifyTestType(const char* typeStr, TestType type) {
    static_assert(isNotEmpty(typeStr), "Test type is empty");
    static_assert(isValidTestType(type), "Test type is invalid");
}
```

In general i think the less Macros the better and google agrees:
"Avoid defining macros, especially in headers; prefer inline functions, enums, and const variables. Name macros with a project-specific prefix. Do not use macros to define pieces of a C++ API.

Macros mean that the code you see is not the same as the code the compiler sees. This can introduce unexpected behavior, especially since macros have global scope."
https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros

Reply:
Not applicable this macros are not ommitable.